docs/apache-airflow/index.rst

```diff
@@ -79,6 +79,49 @@ seen running over time:
 Each column represents one DAG run. These are two of the most used views in Airflow, but there are several
 other views which allow you to deep dive into the state of your workflows.
 
+Customizing Airflow
+===================
+
+There are a number of ways to extend the capabilities of Apache Airflow. Some common methods
+for extending Airflow include:
+
+* Writing :doc:`Custom Operators <howto/custom-operator>`_ : Operators are the building blocks of DAGs in
+  Airflow, and can be used to perform specific tasks within a DAG. By writing custom operators,
+  you can add new functionality to Airflow, such as the ability to connect to and interact
+  with external services, or to perform complex data transformations.
+
+* Writing :doc:`Custom Decorators <howto/create-custom-decorator>`_ : Custom decorators are a way to
+  extend the functionality of the Airflow ``@task`` decorator. By writing custom decorators,
+  you can add new functionality to Airflow, such as the ability to connect to and interact
+  with external services, or to perform complex data transformations.
+
+* Creating custom :doc:`plugins`: Airflow plugins are extensions to the core system that provide additional
+  features and functionality. By creating custom plugins, you can add new capabilities to Airflow,
+  such as adding custom UI components, creating custom macros, creating your own TimeTables,
+  adding Operator Extra Links, Listeners.
+
+* Writing custom :doc:`provider packages <apache-airflow-providers:index>`: Providers are the components
+  of Airflow that are responsible for connecting to and interacting with external services. By writing
+  custom providers, you can add support for new types of external services, such as databases or
+  cloud services, to Airflow.
+
+With all the above methods, you can use the :doc:`public-airflow-api` to write your
+custom Python code to interact with Airflow.
+
+However, you can also extend Airflow without directly integrating your Python code - via the
+:doc:`stable-rest-api-ref` and the :doc:`stable-cli-ref`. These methods allow you to interact with
+Airflow as an external system:
```

Comment:
Would structure this doc starting with the three key points:

1. Extend Python code
2. REST API
3. CLI

And then go deeper into each of the three. This gives the reader an immediate understanding of what's possible instead of having to read through all the options of point nr 1 first.

Reply:
Yes. Done that.